src/somef/extract_software_type.py

```diff
@@ -0,0 +1,315 @@
+    elif check_command_line(path_repo):
+        metadata_result.add_result(constants.CAT_TYPE,
+                                    {
+                                        constants.PROP_VALUE: 'commandline-application',
+                                        constants.PROP_TYPE: constants.STRING                                        
+                                    },
+                                    0.82,
+                                    constants.TECHNIQUE_HEURISTICS)
+
+    elif check_extras(path_repo):
+        metadata_result.add_result(constants.CAT_TYPE,
+                                    {
+                                        constants.PROP_VALUE: 'non-software',
+                                        constants.PROP_TYPE: constants.STRING
+                                    },
+                                    1,
+                                    constants.TECHNIQUE_HEURISTICS)
+    return metadata_result
+
+
+def check_notebooks(path_repo):
+    """Function which checks if the specified repository is a Notebook Application
+       depending on the extensions present and number of notebooks which contain code
+
+       The function checks for presence of more than one code_notebook inside the repo and checks against code 
+       extensions present returning true if both conditions are fulfilled ."""
+    code_notebooks = 0
+    total_files=0
+
+    bad_extensions=False
+    for root, dirs, files in os.walk(path_repo):
+        for file in files:
+            if file.endswith((".ipynb", ".rmd",'.Rmd',".jl")):
+                notebook_path = os.path.join(root, file)
+                try:
+                    if file.endswith(".ipynb"):
+                        if is_notebook_code(notebook_path):
+                            code_notebooks+=1
+                    elif file.endswith(".rmd") or file.endswith('.Rmd'):
+                        if has_code_in_rmd(notebook_path):
+                            code_notebooks += 1
+                except Exception as e:
+                    print(f"Error reading notebook file {notebook_path}: {str(e)}")
+                    pass
+            if file.endswith(constants.code_extensions):
+                bad_extensions=True
+    if code_notebooks>1:
+        return (not bad_extensions)
+
+
+def check_ontologies(path_repo):
+    """Function which detects if repository is an Ontology based on files present
+       and the non-existence of code files"""
+    ontology=False
+    for root, dirs, files in os.walk(path_repo):
+        repo_relative_path = os.path.relpath(root, path_repo)
+        for file in files:
+            file_path = os.path.join(repo_relative_path, file)
+            #print(os.path.join(repo_relative_path,file_path))
+            if file.endswith(constants.code_extensions):
+                return False
+            elif file.endswith(constants.ontology_extensions):
+                if not ontology:
```

Comment:
isn't this redundant? ontology is always False before this

Reply:
I would say no because i want in the code for it to find the first file that represents an ontology not to analyze all of them, so if the code finds in the repo an ontology it would not run the is_file_ontology function for the rest of the ontology extension files. This way it covers the heuristic and also takes less time to complete if a repository contains multiple ontology files

Reply:
ok